### PR TITLE
Back end dagger

### PR DESCRIPTION
**Pull request by Acliu1498**

### CODE REVIEW

File: src/main/kotlin/persistence/repo/Module.kt [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-import persistence.repo.UserRepo
+import javax.inject.Singleton
 
-class TrDatabaseImpl: TrDatabase {
-    private val dataStore: KotlinEntityDataStore<Persistable>
-    private val userRepo: Dao<User>
-    private val languageRepo: Dao<Language>
-    init {
+@Module
+class Module {
```
Comment (sarabiaj) [inline src/main/kotlin/persistence/repo/Module.kt]: Module is too ambiguous of a class name for this

**Comment by sarabiaj** [comment]
Also, dagger shouldn't be used to provide the datastore. Datastore is specific to the implementation of the persistence layer and is being injected into another thing specific to the implementation of the persistence layer, which means we're missunderstanding the purpose of a dependency injection framework.

File: src/main/kotlin/persistence/repo/Module.kt (hunk 1 above)
Comment (mbr4477) [reply src/main/kotlin/persistence/repo/Module.kt]: Got it. That was changed to DaoModule

File: src/main/kotlin/persistence/repo/Component.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package persistence.repo
+
+import dagger.Component
+import dagger.Provides
+import javax.inject.Singleton
+
+@Component(modules = [Module::class])
+@Singleton
+interface Component {
```
[the remark this answers is not in the document]
Comment (mbr4477) [reply src/main/kotlin/persistence/repo/Component.kt]: Changed to DaoComponent

File: src/main/kotlin/persistence/repo/Module.kt (hunk 1 above)
Comment (sarabiaj) [reply src/main/kotlin/persistence/repo/Module.kt]: How about PersistenceModule?

**Pull request closed without merging**